transport/peer.go

```diff
@@ -0,0 +1,48 @@
+package transport
+
+//go:generate mockgen -destination=transporttest/peer.go -package=transporttest go.uber.org/yarpc/transport PeerIdentifier,Peer
+
+// PeerConnectionStatus maintains information about the Peer's connection state
+type PeerConnectionStatus int
+
+const (
+	// PeerAvailable indicates the Peer is available for requests
+	PeerAvailable PeerConnectionStatus = iota + 1
```

Comment:
Order here should be:

0: Unavailable
1: Connecting
2: Available

A zero object should be in the initial state of unavailable.